### ISSUE TITLE
go2rtc 1.9.7 streaming issue for p2p cameras

### ISSUE THREAD

**Issue by andriuszvirblis** [fuatakgun/eufy_security#1249]
After starting P2P Stream and opening Eufy camera entity the video doesnt stream (the window doesnt open completely, only the window header shows).
go2rtc logs shows this error:
12:03:24.761 WRN [rtsp] error="streams: streams: unsupported scheme: T8410P31221846DE" stream=T8410P31221846DE

![image](https://github.com/user-attachments/assets/75ae01ee-a7ba-40c0-b847-0de9e366b61d)

**Comment by oded106e** [comment]
+1

**Comment by saschawj** [comment]
Same here. Initially thought it was a go2rtc issue (which it still may be), however was directed to post it here. 
It may have something to do with HA integrating / messing with WebRTC Camera module for 2024.11.

**Comment by psych0dad** [comment]
This issue was discussed in the go2rtc project.  The maintainer concluded that the issue is rooted in the eufy integration because it creates an invalid configuration. 

https://github.com/AlexxIT/go2rtc/issues/1453#issuecomment-2478365572

A downgrade of the go2rtc version 1.9.6 fixes this issue as a short term solution.

**Comment by philippk75** [comment]
Hello, but do i have to install go2RTC and WEBRTC after the 2014.11 update?

**Comment by rix84** [comment]
> This issue was discussed in the go2rtc project. The maintainer concluded that the issue is rooted in the eufy integration because it creates an invalid configuration.
> 
> [AlexxIT/go2rtc#1453 (comment)](https://github.com/AlexxIT/go2rtc/issues/1453#issuecomment-2478365572)
> 
> A downgrade of the go2rtc version 1.9.6 fixes this issue as a short term solution.

Does anyone know when a hotfix is planned? 
In the meantime, does anyone have a procedure for downgrading go2rtc? 
It seems there's no direct way to do this in HA. 
Thanks.

**Comment by fuatakgun** [comment]
I will check on this weekend

**Comment by z-master42** [comment]
The developer of Go2RTC writes that the source must be empty.

> This is incoming stream. It should have empty source.

> `streams: T8210P812419166F:`

as an example.

**Comment by misc2000** [comment]
> > This issue was discussed in the go2rtc project. The maintainer concluded that the issue is rooted in the eufy integration because it creates an invalid configuration.
> > [AlexxIT/go2rtc#1453 (comment)](https://github.com/AlexxIT/go2rtc/issues/1453#issuecomment-2478365572)
> > A downgrade of the go2rtc version 1.9.6 fixes this issue as a short term solution.
> 
> Does anyone know when a hotfix is planned? In the meantime, does anyone have a procedure for downgrading go2rtc? It seems there's no direct way to do this in HA. Thanks.

Go in HA to Settings->System->Backup and restore from the backup that by default is done when you update a add-on the Version 1.9.6 of go2rtc.

**Comment by misc2000** [comment]
My understanding from the go2rtc developer is also that the entries in the go2rtc config produced automatically from eufy_security? are not right but anyhow did work in past versions. But from version 1.9.6 on this stops working and need to be adapted to the right syntax...

**Comment by fuatakgun** [comment]
I will fix it today.

When this was first developed, it was working fine, something has changed apparently on go2rtc end probably

**Comment by fuatakgun** [comment]
I do not have a working solution until AlexxIT can help me on finding a way to add empty source streams using API.

Please track over here: https://github.com/AlexxIT/go2rtc/issues/1453

Feel free to downgrade go2rtc until this is resolved.

**Comment by profucius** [comment]
> Go in HA to Settings->System->Backup and restore from the backup that by default is done when you update a add-on the Version 1.9.6 of go2rtc.

> Feel free to downgrade go2rtc until this is resolved.

And what about those of us who just joined this project and do not have a previous backup to restore to?

**Comment by fuatakgun** [comment]
Is this not enough answer?

```
I do not have a working solution until AlexxIT can help me on finding a way to add empty source streams using API.
```

**Comment by fuatakgun** [comment]
I believe having only webrtc integration without installing go2rtc would mitigate the issue until I fix it with go2rtc add-on itself.

**Comment by profucius** [comment]
> I believe having only webrtc integration without installing go2rtc would mitigate the issue until I fix it with go2rtc add-on itself.

This doesn't seem to work. I have WebRTC 3.6.0 integration enabled (says "no devices or entities"), with go2rtc addon installed but stopped, and Eufy Security addon 8.0.4 showing all the devices and entities, however camera entities display no camera feed.

**Comment by fuatakgun** [comment]
Configure webrtc and set the go2rtc endpoint as empty in webrtc integration configuration.

This is how it is working in my setup, I am not using go2rtc directly.

Installing an earlier version of add-on is beyond the question of me.

**Comment by profucius** [comment]
I'm not sure what I'm doing wrong then, or if this is the place to ask, but I'm also not sure where to ask so if I may I'll ask here.

I tried your suggestion, deleting the configuration for WebRTC and leaving the URL blank. But I'm still not getting camera feed. I get entity updates for all the other Eufy camera details, but no camera feed itself.

Screenshots of how I have everything setup. Am I doing something wrong?

![1a](https://github.com/user-attachments/assets/5ba029e8-1a20-44a5-91b8-51a357a6a98e)

![2a](https://github.com/user-attachments/assets/c4e6103a-d27a-4dd5-ba32-90ac6a70aec2)

![2b](https://github.com/user-attachments/assets/9790053f-a3b4-403b-b66a-3b7238a161d6)

![3a](https://github.com/user-attachments/assets/0c07afde-873f-4037-8f36-c06e60623ba3)

![3b](https://github.com/user-attachments/assets/30be3845-ffcf-46c1-aefa-dc6d6bd72bf4)

**Comment by z-master42** [comment]
Has the stream been started? This is a battery-powered model, isn't it?

**Comment by fuatakgun** [comment]
issue seems to be fixed by go2rtc, will be published in next version.

if you do not want to wait for it, you can install go2rtc master version (always latest code changes), it is working without any changes on eufy_security side.

**Comment by skrobul** [comment]
> if you do not want to wait for it, you can install go2rtc master version (always latest code changes), it is working without any changes on eufy_security side.

it is not. The incorrect entries are still being created in `go2rtc.yaml`:

```
apollo:/config# cat /config/go2rtc.yaml 
streams:
        T8222P6021172029:
          - T8222P6021172029
apollo:/config# 
```

as soon as the `camera.turn_on` or `eufy_security.start_p2p_stream` is called.
I have compiled today's version of go2rtc that includes @AlexxIT's [commit](https://github.com/AlexxIT/go2rtc/commit/25145f72e56560666e6718740442d16a8c5127bb) :

```
apollo:/config# md5sum /bin/go2rtc /config/go2rtc-1.9.4
e674cb4082a9b3337cad8d853a9f5aea  /bin/go2rtc
e674cb4082a9b3337cad8d853a9f5aea  /config/go2rtc-1.9.4
apollo:/config# /bin/go2rtc --version
go2rtc version 1.9.7 (mod.f1ba5e9) linux/amd64
apollo:/config# 
```

I am using eufy_security integration 8.1.1. The only way to get this to work is by opening the go2rtc webui and deleting this stream. It will then work until next time the camer.turn_on is called


![image](https://github.com/user-attachments/assets/b18d9262-999d-4819-b3b7-b9ae99eafd8d)

**Comment by z-master42** [comment]
I cannot confirm this. It works for me without me having to delete anything.

**Comment by fuatakgun** [comment]
Using master works for me without any manual intervention

**Comment by skrobul** [comment]
Yeah, sorry for the false alarm - it does work for me now after restarting everything - perhaps I had something cached or stale connection somewhere. I'll keep an eye on this and report / investigate if it happens again.

**Comment by skrobul** [comment]
it didn't take long - the issue is back. 

![image](https://github.com/user-attachments/assets/f812b264-d691-4744-95f0-37c1a959be2e)
![image](https://github.com/user-attachments/assets/c7565395-4b31-4c79-be51-6a5d7b43d85c)


@fuatakgun - Any ideas what debug information could I capture to help resolve this?

**Comment by fuatakgun** [comment]
The issue cannot be back, if it works fine.

The issue we have been discussing here is about go2rtc/eufy-security integration compatibility. There was a change on the go2rtc side with `incoming source` streams, it was fixed on the master branch. Whenever you use the master branch, you are back to the previous state (which could be somehow broken somehow working)

Unsupported scheme warning was always there but it is not a blocking issue just a warning.

Depending on your case, either open a new issue with debug logs or let's continue here.

**Comment by skrobul** [comment]
> The issue cannot be back, if it works fine once.

Well it's clearly intermittent - once it stops working it stops, and the symptoms that are visible are exactly as described in this issue here that's why I thought it's related.


> There was a change on the go2rtc side with incoming source streams, it was fixed on the master branch. Whenever you use the master branch, you are back to the previous state (which could be somehow broken somehow working)

Again, I am still using the exactly same version of go2rtc as I was using 4 hours and 20 hours ago - specifically the `f1ba5e9` commit which contains fixes for the `incoming source` stuff.

When I posted my previous comment, any attempt to open p2p stream resulted in dreaded `mse: unsupported scheme [SERIAL]`.  Debug logs for `eufy_security` did not show anything out of ordinary - just the usual `delete stream` followed by `create stream`.


> Unsupported scheme warning was always there but it is not a blocking issue just a warning.

oh okay understood - I wrongly assumed that's the problem because that's the very last thing that's visible from the end user perspective.

I have now restarted hass to enable debug logs for `custom:webrtc` and the video feed is working again. When it breaks again, hopefully I will have more information to go on. 


> Depending on your case, either open a new issue with debug logs or let's continue here.

At this stage I don't even know if this is right place to open the issue - if the unsupported scheme isn't a problem. Based on my [previous comment](https://github.com/fuatakgun/eufy_security/issues/1249#issuecomment-2525775954) it looks like `eufy_security` integration registers the stream `producer` and pushes some data to it, but there is no `consumers` registered so it may be a problem on the go2rtc/webrtc card side - if that's the case, sorry for spamming you :) .
